File: _inc/_DEPRECATED_django/erp_prestech_backend/app/Mail/common_email_template.py
from django.core.mail import EmailMultiAlternatives
from django.template.loader import render_to_string

class CommonEmailTemplate:
    def __init__(self, template, settings):
        """
        :param template: A dict with keys "from", "subject", and "content"
        :param settings: A dict with key "mail_from_address"
        """
        self.template = template
        self.settings = settings

    def build(self):
        """
        Builds an EmailMultiAlternatives message using a markdown template.
        Assumes there is a template file at "email/common_email_template.md"
        in your Django templates directory.
        """
        # Construct the sender: using the display name from the template and the email from settings
        from_email = f"{self.template['from']} <{self.settings['mail_from_address']}>"
        subject = self.template['subject']
        # Render the email body using a markdown template with the provided content
        context = {'content': self.template['content']}
        body = render_to_string("email/common_email_template.md", context)
        # Create the email message instance (recipients can be set later)
        email_message = EmailMultiAlternatives(
            subject=subject,
            body=body,
            from_email=from_email,
            to=[],  # to be set when sending
        )
        return email_message

File: _inc/laravel/app/Mail/CommonEmailTemplate.php
<?php

namespace App\Mail;

use Illuminate\Bus\Queueable;
use Illuminate\Mail\Mailable;
use Illuminate\Queue\SerializesModels;

class CommonEmailTemplate extends Mailable
{
    use Queueable, SerializesModels;

    public $template;
    public $settings;

    /**
     * Create a new message instance.
     *
     * @return void
     */
    public function __construct($template, $settings)
    {

        $this->template = $template;
        $this->settings = $settings;
    }

    /**
     * Build the message.
     *
     * @return $this
     */
    public function build()
    {
        $fromAddress = is_array($this->settings) ? ($this->settings['mail_from_address'] ?? config('mail.from.address')) : config('mail.from.address');
        $fromName = is_object($this->template) ? ($this->template->from ?? config('mail.from.name')) : config('mail.from.name');
        $subject = is_object($this->template) ? ($this->template->subject ?? '') : '';
        $content = is_object($this->template) ? ($this->template->content ?? '') : '';
        return $this->from($fromAddress, $fromName)->markdown('email.common_email_template')->subject($subject)->with('content', $content);
    }
}
